src/py/flwr/server/fleet/grpc_bidi/ins_scheduler.py

```diff
@@ -100,16 +101,10 @@ def _worker(
         )
 
         # Step 3: wrap FitRes in a ClientMessage in a Task in a TaskRes
-        task_res = TaskRes(
-            task_id="",  # Will be created and set by the State
-            group_id=task_ins.group_id,
-            workload_id=task_ins.workload_id,
-            task=Task(
-                producer=Node(node_id=client_proxy.node_id, anonymous=False),
-                consumer=Node(node_id=0, anonymous=True),
-                legacy_client_message=client_message_proto,
-                ancestry=[task_ins.task_id],
-            ),
+        task_res = configure_task_res(
+            TaskRes(task=Task(legacy_client_message=client_message_proto)),
+            task_ins,
+            Node(node_id=0, anonymous=True),
```

Comment:
Don't we need `Node(node_id=client_proxy.node_id, anonymous=False),` here? The `consumer` field needs to be `Node(node_id=0, anonymous=True),`

Reply:
yes. changed to `Node(node_id=client_proxy.node_id, anonymous=False)`